### PR TITLE
Add restart identity to truncate SQL query for pgsql

### PR DESCRIPTION
**Pull request by aivus**
- fix tests

|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no?
| Fixed issues | #2882

#### Summary

PgSQL and MySQL have different behaviour now.
This PR provides a way to have same behaviour on truncate commandю

MySQL resets PK automatically when truncate executed:
> Any AUTO_INCREMENT value is reset to its start value. This is true even for MyISAM and InnoDB, which normally do not reuse sequence values.

https://dev.mysql.com/doc/refman/5.7/en/truncate-table.html

PgSQL requires additional parameters to follow this behaviour:
https://www.postgresql.org/docs/9.2/static/sql-truncate.html

### CODE REVIEW

**Comment by aivus** [comment]
Any advice how to fix CS?

```
FILE: lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 1098 | ERROR | [x] Equals sign not aligned with surrounding assignments;
      |       |     expected 13 spaces but found 1 space
      |       |     (Generic.Formatting.MultipleStatementAlignment.NotSame)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

**Comment by Majkl578** [comment]
The error tells you what's wrong:

> Equals sign not aligned with surrounding assignments; expected 13 spaces but found 1 space

You need to align `=` with surrounding statement(s), in this case with `$tableIdentifier =` above.

**Comment by aivus** [comment]
Looks weird, actually. Hope this commit will fix it.

**Comment by morozov** [comment]
@aivus please add a functional test. The test should pass on all platforms which implement autoincrement natively.

File: tests/Doctrine/Tests/DBAL/Functional/ResetAutoincrementTest.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class ResetAutoincrementTest extends DbalFunctionalTestCase
+{
+    public function testAutoincrementResetsOnTruncate()
+    {
+        $table = new Table('autoincremented_table');
+        $table->addColumn('id', 'integer', ['autoincrement' => true]);
+        $table->addColumn('test_int', 'integer');
+        $table->setPrimaryKey(['id']);
+
+        /** @var AbstractSchemaManager $sm */
```
[the remark this answers is not in the document]
Comment (aivus) [reply tests/Doctrine/Tests/DBAL/Functional/ResetAutoincrementTest.php]: Was needed before. Removed.

File: tests/Doctrine/Tests/DBAL/Functional/ResetAutoincrementTest.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        $this->connection->exec($this->connection->getDatabasePlatform()->getTruncateTableSQL('autoincremented_table'));
+
+        $this->connection->insert('autoincremented_table', ['test_int' => 4]);
+        $lastId = $this->getIdByTestInt(4);
+        $this->assertEquals(1, $lastId);
+    }
+
+    protected function getIdByTestInt(int $whereTestInt)
+    {
+        $row = $this->connection->fetchAssoc('SELECT id FROM autoincremented_table WHERE test_int = ?', [$whereTestInt]);
```
[the remark this answers is not in the document]
Comment (aivus) [reply tests/Doctrine/Tests/DBAL/Functional/ResetAutoincrementTest.php]: Fixed

File: lib/Doctrine/DBAL/Platforms/SqlitePlatform.php [hunk 3]
```diff
@@ -535,9 +535,12 @@ public function getName()
...
     {
         $tableIdentifier = new Identifier($tableName);
-        $tableName       = str_replace('.', '__', $tableIdentifier->getQuotedName($this));
+        $quotedTableName = str_replace('.', '__', $tableIdentifier->getQuotedName($this));
 
-        return 'DELETE FROM ' . $tableName;
+        $deleteSql        = 'DELETE FROM ' . $quotedTableName;
+        $resetSequenceSql = 'DELETE FROM sqlite_sequence WHERE name = ' . $this->quoteStringLiteral($tableName);
+
+        return $deleteSql . '; ' . $resetSequenceSql;
```
Comment (morozov) [inline lib/Doctrine/DBAL/Platforms/SqlitePlatform.php]: Usually, DB drivers for PHP do not allow multiple statements in a call for security reasons. Is SQLite an exception or all PDO-based drivers work like this?

**Comment by aivus** [comment]
Didn't found the way to restart identity for `ibm_db2`

I'm not going to continue work on this PR. Closing it.

**Pull request closed without merging**